GoogleCloudPlatform/datacatalog-connectors-rdbms
# Extend MySQL and PostgreSQL connectors

**- What I did**

1. Added table size, in MB, to the Tag Template 
2. Implemented scraping of the table_size for MySQL and PostgreSQL
3. Implemented scraping of create_time and update_time for MySQL tables
4. Introduced user config (see "ingest_cnfg.yaml"), where a user can toggle options for updating metadata and scraping additional metadata
5. Implemented metadata update with ANALYZE command for PostgreSQL andMySQL
6. Implemented timing of the ANALYZE command for user to know how long it takes
7. Implemented (optional) scraping of num_rows in MySQL and PostgreSQL tables

**- How I did it**

According to design doc

**- How to verify it**

1. Create two databases: MySQL and PostgreSQL
2. Populate them with random data via metadata generator ([Postgres](https://github.com/mesmacosta/google-datacatalog-connectors-profiler/tree/master/postgresql/metadata_generator), [MySQL](https://github.com/mesmacosta/google-datacatalog-connectors-profiler/tree/master/mysql/metadata_generator)) 
2.1 (Optional) To get number of rows counted, add some data to the newly created tables, as the metadata generation script leaves the tables empty.
3. Set up a project on GCP.
4. Run connectors ([Postgres](https://github.com/GoogleCloudPlatform/datacatalog-connectors-rdbms/tree/master/google-datacatalog-postgresql-connector), [MySQL](https://github.com/GoogleCloudPlatform/datacatalog-connectors-rdbms/tree/master/google-datacatalog-mysql-connector)) 
4.1 You can try running them with different user configurations. Configurations are stored in ingest_cfg.yaml file. 
5. In DataCatalog, Tag Template "XY Table - Metadata", you should see a new field - "table_size_MB". In Table Tags you should see value for this new field. If there are any data in your tables, you should see the num_rows field of Table Tags populated as well. Besides, there should be timestamps in 'Created' and 'Last Modified' fields of entry details for MySQL tables.

**- Description for the changelog**
Extend PostgreSQL and MySQL connectors
